vscode-extension/src/extension.ts

```diff
@@ -325,3 +642,92 @@ async function checkPip() {
     });
   });
 }
+
+async function shareAIConfig(
+  context: vscode.ExtensionContext,
+  aiconfigEditorManager: AIConfigEditorManager
+) {
+  // Get the current active aiconfig editor
+  const activeEditor = aiconfigEditorManager.getActiveEditor();
+  if (!activeEditor) {
+    vscode.window.showErrorMessage(
+      "No AIConfig file is currently open in the editor. Please open the file you want to share and try again."
+    );
+    return;
+  }
+
+  const fileName: string = activeEditor.document.fileName;
+  const sanitizedFileName: string = sanitizeFileName(fileName);
+
+  // Use empty accessKeyId and secretAccessKey so bucket can be accessed publicly
+  AWS.config.update({
+    accessKeyId: "",
+    secretAccessKey: "",
+  });
+  const s3 = new AWS.S3();
+
+  const randomPath = Math.round(Math.random() * 10000);
+  const bucket: string = "lastmileai.aiconfig.public";
+  const uploadKey: string = `aiconfigs/${getTodayDateString()}/${randomPath}/${sanitizedFileName}`;
+  const configString = activeEditor.document.getText();
+
+  // TODO: Will also need to check for yaml files and change the contentType accordingly
```

Comment:
For sharing YAML, we'll also need to update the readonly implementation to parse to JSON for rendering it